jupyterhub/services/service.py

```diff
@@ -189,6 +190,13 @@ class Service(LoggingConfigurable):
         """
     ).tag(input=True)
 
+    allowed_roles = List(
```

Comment:
Maybe let's call this `oauth_roles`? I think it might be a little clearer to have oauth in the name to communicate that these are the allowed (and requested by default) roles for this service *when it functions as an oauth client*.

Then it'll be `add_oauth_client(allowed_roles=service.oauth_roles)`

Reply:
Good idea, I'll fix that